### DISCUSSION TITLE
[논의] UUID version?

### DISCUSSION THREAD

**Discussion by kimhyun5u** [woowa-techcamp-2024/Team1-3K1K#45] [General]
### 배경

```bash
Vendor 와 Customer 의 id 값을  uuid 로 통일하자는 의견이 나오고 조사를 진행하던 중 Java에서 기본으로 제공하는 UUID v4는 시간 정보가 안 들어간다는 사실을 알게 되었다.
이에 따라 정렬을 다른 컬럼으로 해야한다.
UUID v1 와 UUID v7 은 시간 순이다.
UUID v1은 MAC 주소가 들어가서 보안에 취약하고 UUID v7은 표준화가 아직 덜 진행되었다.
```

### 세부 주제

```bash
1.UUID의 어떤 버전을 사용할지
```

**Comment by kimhyun5u** [comment]
정보: Hibernate 6 이상부터는 GenerateType.UUID 는 기본 UUID v4 를 지원한다.

**Comment by kimhyun5u** [comment]
제 생각에는 그냥 Hibernate 기본 바전인 UUID v4를 사용해도 될 것 같다는 생각이 드네요!

**Comment by Hyeon-Uk** [comment]
좋습니다! 제일 안정적인 v4를 사용하고, 정렬은 다른 컬럼을 이용해서 해도 괜찮을것같네요!

**Comment by kimhyun5u** [comment]
문제: UUID 로만 분류하면 더미데이터만들 때 복잡성이 발생함

**Reply by kimhyun5u** [reply]
그래서 auto_increment sequence_id 를 제시함

**Reply by kimhyun5u** [reply]
select id from question order by RAND() limit 500; 로 위의 문제가 해결됨을 알 수 있음

**Comment by june-777** [comment]
저도 UUID 는 v4를 사용하고, 정렬이 필요하면 auto_increment의 기능을 활용하여 별도의 컬럼을 활용하면 좋을 것 같습니다!

**Comment by kimhyun5u** [comment]
결론: UUID 를 id 로 하고 Hibernate 6.0 이후부터 기본으로 제공하는 UUID v4 를 사용.
